### ISSUE TITLE
[Cross-Plugins] Does neo-tree.nvim move the cursor to the first line after the first file is open?

### ISSUE THREAD

**Issue by nyngwang** [nvim-neo-tree/neo-tree.nvim#633]
This problem is related to fzf-lua https://github.com/ibhagwan/fzf-lua/issues/572. If I remove neo-tree.nvim the linked problem disappeared. Since I have no clue about what happened so I need your help!

**Comment by nyngwang** [comment]
I just confirmed that the problem can be solved (but I'm not sure about the reason) by moving the cursor out of neo-tree buffer before calling any fzf-lua APIs.

**Comment by cseickel** [comment]
If the cursor is in the neo-tree window when you call a fzf api, it will probably try to open whatever file you selected within the Neo-tree window. Neo-tree will then detect that it's window was stolen, and move that intruder to a new window and then reclaim the window that Neo-tree originally was in. 

This normally happens so fast that you can't see it, but there is the opportunity for problems in this little asynchronous dance.

**Comment by nyngwang** [comment]
@cseickel Is it possible to optimize the process? Say I have the option `close_if_last_window ` as to `false`. Then neo-tree is the only window/buffer. In that case I have to call fzf-lua's command inside neo-tree. Sometimes it failed and result in some strange error(I forgot its content), and at that moment while the selected buffer did get open, the buffer itself cannot be modified and/or the filetype is `neo-tree`.

**Comment by figsoda** [comment]
I am experiencing the same thing with telescope, is it possible to configure the behavior to `open_drop`?

**Comment by nyngwang** [comment]
@figsoda What is `open_drop`

**Comment by pysan3** [comment]
> I am experiencing the same thing with telescope

I am not very sure but [this discussion](https://github.com/nvim-telescope/telescope.nvim/issues/559) might be related to this problem?

**Comment by cseickel** [comment]
> I am not very sure but [this discussion](https://github.com/nvim-telescope/telescope.nvim/issues/559) might be related to this problem?

I finally figured out why Gitsigns does not always work! It's when I open a file via a Telescope method! I may have to investigate and try to fix that...

As far as Neo-tree is concerned though, all the problems arise from deflecting other buffers that try to steal its window. I think the ultimate solution is if you have Neo-tree as a sidebar, it should prevent itself from ever becoming the only open window. I just did something similar for when `close_if_last_window` activates and there is a modified unnamed buffer that will cause an error. It will then create a window in the right location and re-open that buffer.

I can take that logic and just add an empty new buffer whenever the last window is closed. This will preserve the layout and allow you to open new files into that buffer when you are using fzf or telescope.

The downside is that you would have to use `:qa` now if you want to close neovim completely. That is how I close it anyhow, personally.

**Comment by figsoda** [comment]
> What is the behavior of open_drop?

@nyngwang For me, it keeps the position as if I opened the window normally (not in neo-tree buffer)

> I am not very sure but https://github.com/nvim-telescope/telescope.nvim/issues/559 might be related to this problem?

@pysan3 Seems like it might be the same problem I'm experiencing, however I have folds turned off and I am not expriencing the treesitter part, only not having the cursor position restored

**Comment by cseickel** [comment]
> To @cseickel:
> 
> > [...]
> > The downside is that you would have to use `:qa` now if you want to close neovim completely.
> 
> Is this true even if I (use toggle to) close any neo-tree buffer before I want to quit neovim? 

No, this code would only run when Neo-tree finds itself as the last remaining window, so it would have no effect if Neo-tree is closed.

**Comment by nyngwang** [comment]
@cseickel:

> No, this code would only run when Neo-tree finds itself as the last remaining window [...]

Then I hope this fix can be merged as soon as possible!

**Comment by nyngwang** [comment]
The problem can be solved by the following workaround:


```lua
vim.keymap.set('n', '<C-f>', function ()
  if vim.bo.filetype == 'neo-tree' then -- from here.
    if #vim.api.nvim_tabpage_list_wins(0) == 1 then vim.cmd('Neotree close') return end
    vim.cmd('wincmd h')
  end -- to here.
  require('fzf-lua').live_grep_glob({ actions = { ['ctrl-f'] = 'abort' } }) -- before calling fzf-lua.
end, NOREF_NOERR_TRUNC)
```

without these lines the problem still persists:

![image](https://user-images.githubusercontent.com/24765272/207733093-bac8eb13-122f-4252-9d19-8d241117693e.png)

and if I press `<C-o>`, i.e. trying to restore my cursor step-by-step, I got this:

![image](https://user-images.githubusercontent.com/24765272/207733330-67a52c4a-4ebf-44fc-96a5-6b4ca60e13ab.png)

this indeed confirms what you have said in another issue.

**Comment by edevil** [comment]
I think I'm hitting this same issue while opening a target result from fzf while neo-tree had focus, the cursor doesn't go the matched line. Demo here: https://github.com/ibhagwan/fzf-lua/issues/1684

Was a solution found for this issue?
